geode-management/src/test/java/org/apache/geode/management/configuration/RegionTest.java

```diff
@@ -64,6 +64,72 @@ public void correctJson() throws Exception {
     mapper.readValue(json2, Region.class);
   }
 
+  @Test
+  public void correctJsonWithExpiriations() throws Exception {
+    String expireJson = "{\n"
+        + "  \"name\": \"region1\",\n"
+        + "  \"type\": \"PARTITION\",\n"
+        + "  \"expirations\": [\n"
+        + "    {\n"
+        + "      \"type\": \"ENTRY_IDLE_TIME\",\n"
+        + "      \"timeInSeconds\": 3600,\n"
+        + "      \"action\": \"DESTROY\"\n"
+        + "    }\n"
+        + "  ]\n"
+        + "}\n";
+    regionConfig = mapper.readValue(expireJson, Region.class);
+    assertThat(regionConfig.getName()).isEqualTo("region1");
+    assertThat(regionConfig.getType()).isEqualTo(RegionType.PARTITION);
+    assertThat(regionConfig.getExpirations()).isNotNull();
+    assertThat(regionConfig.getExpirations().size()).isEqualTo(1);
+
+    Region.Expiration expiration = regionConfig.getExpirations().get(0);
+    assertThat(expiration.getType()).isEqualTo(Region.ExpirationType.ENTRY_IDLE_TIME);
+    assertThat(expiration.getTimeInSeconds()).isEqualTo(3600);
+    assertThat(expiration.getAction()).isEqualTo(Region.ExpirationAction.DESTROY);
+
+    String json2 = mapper.writeValueAsString(regionConfig);
+    System.out.println(json2);
+    assertThat(json2).contains("\"type\":\"PARTITION\"");
+    assertThat(json2).contains("\"name\":\"region1\"");
+    assertThat(json2).contains("\"type\":\"ENTRY_IDLE_TIME\"");
+    assertThat(json2).contains("\"timeInSeconds\":3600");
+    assertThat(json2).contains("\"action\":\"DESTROY\"");
+
+    mapper.readValue(json2, Region.class);
+  }
+
+  @Test
+  public void correctJsonWithEviction() throws Exception {
+    String evictJson = "{\n"
+        + "  \"name\": \"region1\",\n"
+        + "  \"type\": \"PARTITION\",\n"
+        + "  \"eviction\": {\n"
+        + "    \"type\": \"ENTRY_COUNT\",\n"
```

Comment:
if the json string has no "type":"ENTRY_COUNT", would the type being set correctly?

Reply:
Yes, upon conversion we assume that given that we have an entryCount value that the type is `ENTRY_COUNT` and we save it as such in the cluster configuration.